packages/design-toolkit/src/components/coordinate-field/index.tsx

```diff
@@ -248,61 +250,96 @@ export function CoordinateField({ ref, ...props }: CoordinateFieldProps) {
                   }
-                  placeholder={config.placeholder}
-                  maxLength={config.maxLength}
-                  pad={config.pad}
-                  className={clsx(styles.segment, classNames?.segment)}
-                  isDisabled={isDisabled}
-                  allowedChars={config.allowedChars}
-                  segmentRef={focus.segmentRefs[editableIndex]}
-                  segmentIndex={editableIndex}
-                  totalSegments={state.editableSegmentConfigs.length}
-                  ariaLabel={getSegmentLabel(format, editableIndex)}
-                />
-              );
-            })}
+
+                  const editableIndex = state.segmentConfigs
+                    .slice(0, configIndex)
+                    .filter((c) => c.type !== 'literal').length;
+
+                  acc[currentGroupIndex]?.push(
+                    <CoordinateSegment
+                      key={`${format}-segment-${editableIndex}`}
+                      value={state.segmentValues[editableIndex] || ''}
+                      onChange={(newValue) =>
+                        state.handleSegmentChange(editableIndex, newValue)
+                      }
+                      onFocus={() =>
+                        focus.setFocusedSegmentIndex(editableIndex)
+                      }
+                      onBlur={() => {
+                        focus.setFocusedSegmentIndex(-1);
+                        state.flushPendingValidation();
+                      }}
+                      onKeyDown={(e) =>
+                        focus.handleSegmentKeyDown(editableIndex, e)
+                      }
+                      onAutoAdvance={() =>
+                        focus.focusNextSegment(editableIndex)
+                      }
+                      onAutoRetreat={() =>
+                        focus.focusPreviousSegment(editableIndex)
+                      }
+                      placeholder={config.placeholder}
+                      maxLength={config.maxLength}
+                      pad={config.pad}
+                      className={clsx(styles.segment, classNames?.segment)}
+                      isDisabled={isDisabled}
+                      allowedChars={config.allowedChars}
+                      segmentRef={focus.segmentRefs[editableIndex]}
+                      segmentIndex={editableIndex}
+                      totalSegments={state.editableSegmentConfigs.length}
+                      ariaLabel={getSegmentLabel(format, editableIndex)}
+                    />,
+                  );
+
+                  return acc;
+                },
+                [[]],
+              )
+              .map((group, groupIndex) => (
+                <div
+                  key={`group-${
+                    // biome-ignore lint/suspicious/noArrayIndexKey: intentional
+                    groupIndex
+                  }`}
+                >
+                  {group}
+                </div>
+              ))}
```

Comment:
could avoid unnecessarily nesting inlines in an extra div by conditional here:

```suggestion
.map((group, groupIndex) =>
  variant === 'stacked' ? (
    <div
      key={`${format}-group-${groupIndex}`}
    >
      {group}
    </div>
  ) : (
    <Fragment key={`${format}-group-${groupIndex}`}>{group}</Fragment>
  )
)
```